### ISSUE TITLE
PLEASE HELP!! ALT key and Windows key Issue

### ISSUE THREAD

**Issue by finaln1** [Genymobile/scrcpy#5318]
Hello.

I can't send the ALT key and Windows key of the keyboard connected to the PC to the smartphone while connected to the smartphone via scrcpy in Windows 10.

When I press the ALT key, it responds on the PC. The Windows key also responds on the PC, not the smartphone.

I set --keyboard=uhid, but it's the same problem.

--fullscreen didn't solve the problem either.

Please help.

**Comment by rom1v** [comment]
Try with this binary (replace in the v2.7 release):

- [`scrcpy.exe`](https://tmp.rom1v.com/scrcpy/5318/1/scrcpy.exe) <sub>`SHA-256: 54673620d62ee888f0e32624f3acf3617fa6a96f25e17dfd45b8ee21654a798`</sub>

Execute:

```
scrcpy --shortcut-mod=rctrl
```

(for example, so that the shortcut mod is the right ctrl key)

Now, you should be able to use <kbd>Alt</kbd> and <kbd>Super</kbd>.

**Comment by rom1v** [comment]
Yes, I think the mouse capture key should not be hardcoded to LAlt, LSuper or RSuper. I think it may be the same as the shortcut mod key (so if `--shortcut-mods=rctrl`, mouse capture/uncapture would be performed by rctrl). What do you think?

**Comment by rom1v** [comment]
> Even if I use --shortcut-mods=rctrl, if I use it with --mouse=uhid, I can't use the ALT key

Sure, but would that behavior (using the same key as shortcut mods for the mouse capture key) be ok to you?

**Comment by finaln1** [comment]
> > Even if I use --shortcut-mods=rctrl, if I use it with --mouse=uhid, I can't use the ALT key
> 
> Sure, but would that behavior (using the same key as shortcut mods for the mouse capture key) be ok to you?

Yes, it would be nice if I could return the control of the mouse to the PC when I used the MOD Key (ex: rctrl) alone or entered it in combination with another key.
(Because rctrl is typically rarely used.)

**Comment by rom1v** [comment]
Please test #5322.

**Comment by finaln1** [comment]
Hello, I'm telling you what I tested.

1. When I set RCTRL to Mod, when I press RCTRL, mouse control normally goes over to PC.

2. When I press the Super key, it is transmitted normally to the smartphone, but at the same time, the Super key of the PC is also pressed, and mouse control is transferred to the PC.
- I find this part uncomfortable. Is it possible to improve it?

3. When I press the ALT key, it normally passes to my smartphone. But when I press ALT+TAB, ALT+TAB works on my PC.
- I use my smartphone to remotely control other PCs, so I hope this part can be modified as well.

Thank you for your kindness.

**Comment by rom1v** [comment]
Thank you for your feedback.

> 1. When I set RCTRL to Mod, when I press RCTRL, mouse control normally goes over to PC.

:+1:

> 2. When I press the Super key, it is transmitted normally to the smartphone,

:+1:

> but at the same time, the Super key of the PC is also pressed, and mouse control is transferred to the PC.

I cannot reproduce on Linux. I guess the Windows environment captures Super for its own purpose (opening the "start menu" I guess?). It's weird that it both captures and send the event to the application though. Not sure I can do something about it with SDL.

> When I press the ALT key, it normally passes to my smartphone.

:+1:

> But when I press ALT+TAB, ALT+TAB works on my PC.

Yes, the Windows environment / window manager captures Alt+tab. There is a SDL hint ([SDL_HINT_ALLOW_ALT_TAB_WHILE_GRABBED](https://wiki.libsdl.org/SDL2/SDL_HINT_ALLOW_ALT_TAB_WHILE_GRABBED)), but whatever its value, I can't prevent Alt+tab to be captured by the system.

At least, #5322 fixes forwarding of Alt and Super as expected with SDK keyboard (consistent with UHID/AOA keyboards). After that there are other issues because the system may also capture these keys.
